Utils/PluginActivator/source/COMRPCStarter.cpp

```diff
@@ -0,0 +1,149 @@
+ * @param[in]   retryDelayMs    Delay in ms between retry attempts
+ *
+ * @return True if plugin successfully activated, false if failed to activate
+ */
+bool COMRPCStarter::activatePlugin(const int maxRetries, const int retryDelayMs)
+{
+    // Attempt to open the plugin shell
+    bool success = false;
+    int currentRetry = 1;
+
+    PluginHost::IShell* shell = nullptr;
+
+    while (!success && currentRetry <= maxRetries) {
+        LOG_INF(mPluginName.c_str(), "Attempting to activate plugin - attempt %d/%d", currentRetry, maxRetries);
+
+        auto start = std::chrono::steady_clock::now();
+
+        if (!shell) {
+            shell = mClient->Open<PluginHost::IShell>(mPluginName, ~0, RPC::CommunicationTimeOut);
+        }
+
+        // We could not open IShell for some reason - Thunder doesn't give an error about why this might have failed
+        // Normally this is because either:
+        //  a) Thunder isn't running or we can't connect to /tmp/communicator
+        //  b) A plugin with the specified callsign does not exist
+        if (!shell) {
+            LOG_ERROR(mPluginName.c_str(), "Failed to open IShell interface for %s", mPluginName.c_str());
+            currentRetry++;
+
+            // Must close the connection before we retry, since even if shell is null the underlying
+            // connection will still have been opened
+            mClient->Close(RPC::CommunicationTimeOut);
+
+            std::this_thread::sleep_for(std::chrono::milliseconds(retryDelayMs));
+            continue;
+        };
+
+        // Connected to Thunder successfully and got the plugin shell
+        if (shell->State() == PluginHost::IShell::ACTIVATED) {
+            LOG_INF(mPluginName.c_str(), "Plugin is already activated - nothing to do");
+            success = true;
+        } else {
+            // Will block until plugin is activated
+            uint32_t result = shell->Activate(PluginHost::IShell::REQUESTED);
+
+            auto end = std::chrono::steady_clock::now();
+            auto duration = std::chrono::duration_cast<std::chrono::milliseconds>(end - start);
+
+            if (result != Core::ERROR_NONE) {
+                if (result == Core::ERROR_PENDING_CONDITIONS) {
+                    // Ideally we'd print out which preconditions are un-met for debugging, but that data is not exposed through the IShell interface
+                    LOG_ERROR(mPluginName.c_str(), "Failed to activate plugin due to unmet preconditions after %ldms", duration.count());
+                } else {
+                    LOG_ERROR(mPluginName.c_str(), "Failed to activate plugin with error %u (%s) after %ldms", result, Core::ErrorToString(result), duration.count());
+                }
+                // Try activation again up until the max number of retries
+                currentRetry++;
+                std::this_thread::sleep_for(std::chrono::milliseconds(retryDelayMs));
+                continue;
```

Comment:
Minor, for maintainability the coding guidelines we adpoted do not promote the usage of the keyword continue. Try finding the keyword in the current code base ;-)